src/Event.cc

```diff
@@ -199,17 +241,46 @@ void Event::_constructor( const std::string& runId, const PropertySet& ps) {
 
     // _pubTime is filled in on publish and is ignored in the passed PropertySet
     _psp->set(PUBTIME, (long long)0);
+
+    if (filterable.nameCount() > 0) {
+        vector<std::string> names = filterable.names();
+        unsigned int i;
+        for (i = 0; i < names.size(); i++) {
+            _keywords.insert(names[i]);
+        }
+
+        _psp->combine(filterable.PropertySet::deepCopy());
+    }
 }
 
 void Event::populateHeader(cms::TextMessage* msg)  const {
-    msg->setStringProperty(TYPE, _psp->get<std::string>(TYPE));
-    msg->setLongProperty(EVENTTIME, _psp->get<long long>(EVENTTIME));
-    msg->setStringProperty(HOSTID, _psp->get<std::string>(HOSTID));
-    msg->setStringProperty(RUNID, _psp->get<std::string>(RUNID));
-    msg->setStringProperty(STATUS, _psp->get<std::string>(STATUS));
-    msg->setStringProperty(TOPIC, _psp->get<std::string>(TOPIC));
-    msg->setLongProperty(PUBTIME, _psp->get<long long>(PUBTIME));
+    set<std::string>::iterator keyIterator;
+    for (keyIterator = _keywords.begin(); keyIterator != _keywords.end(); keyIterator++) {
+        std::string name = *keyIterator;
+        const std::type_info &t = _psp->typeOf(name);
+        if (t == typeid(bool))
+            msg->setBooleanProperty(name, _psp->get<bool>(name));
+        else if (t == typeid(short))
+            msg->setShortProperty(name, _psp->get<short>(name));
+        else if (t == typeid(int))
+            msg->setIntProperty(name, _psp->get<int>(name));
+        else if (t == typeid(long))
+            msg->setLongProperty(name, _psp->get<long>(name));
+        else if (t == typeid(long long))
+            msg->setLongProperty(name, _psp->get<long long>(name));
+        else if (t == typeid(double))
+            msg->setDoubleProperty(name, _psp->get<double>(name));
+        else if (t == typeid(float))
+            msg->setFloatProperty(name, _psp->get<float>(name));
+        else if (t == typeid(std::string))
+            msg->setStringProperty(name, _psp->get<std::string>(name));
+        else {
+            std::string msg("Data type represented in "+name+" is not permitted in event header");
+            throw LSST_EXCEPT(pexExceptions::RuntimeError, msg);
```

Comment:
You shouldn't; this is a straightforward case of invoking `operator+(const char *, std::string)` then `operator+(std::string, const char*)`, both of which are overloaded by `std::string`.